napari/_tests/test_conftest_fixtures.py

```diff
@@ -0,0 +1,37 @@
+import time
+
+import pytest
+from qtpy.QtCore import QThread, QTimer
+
+
+class _TestThread(QThread):
+    def run(self):
+        time.sleep(1)
```

Comment:
1 second is a long time for a unit test. Currently, we only execute this sleep twice and we're unlikely to expand this test much in the future, so it's not a big problem.

But ideally I'd like to see use of locks here to both speed these test up and explicitly control concurrent execution (i.e. rather than rely on timing). To do that, you could define a Python lock as an attribute of `_TestThread` then acquire that in `run` and in the part of the calling test where you want to prevent `run` from finishing (e.g. for `test_qthread_running` you would add a `with t.lock:` context manager for `t.start()` and `assert t.isRunning()`.

Reply:
I rewrite in on `mutex` It looks better now?

Reply:
Looks great!

Reply:
My only small concern is what happens when any lines in between `lock` and `unlock` fail either now or in the future (i.e. if that happens, is it possible for the test to wait forever in that case? Or is there something else that protects us?

If not, I'd prefer a scoped mutex/lock (e.g. with `QMutexLocker`) or asserting the return value of `QMutex.tryLock` with a timeout in `TestThread.run`